nc-news_/src/Components/ArticleList/ListProvider.jsx

```diff
@@ -0,0 +1,48 @@
+import apiClient from "../../api";
+import { useEffect, useState } from "react";
+import ArticleCard from "./ArticleCard";
+
+const ListProvider = () => {
+  const [articleList, setArticleList] = useState([]);
+  const [page, setPage] = useState(1);
+  const [articleCount, setArticleCount] = useState(0);
+  const totalPages = Math.ceil(articleCount / 10);
+
+  const nextPageHandler = () => {
+    setPage(page + 1);
+  };
+  const previousPageHandler = () => {
+    setPage(page - 1);
+  };
```

Comment:
Really great that you have pagination here however, be careful with adding code for additional tasks, remember that tasks should be built with a single focus in mind and commits related to a task should only include relevant code needed to add that particular piece of functionality.